### PR TITLE
ci: Use protocol-version to tag prover images

### PR DESCRIPTION
**Pull request by yorik**
## What ❔

Tag prover packages with protocol version.

Removes unneeded dependencies of `prover-version`.
Remove outdated images from builder:
- prover
- prover-v2
- circuit-synthesizer

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.

### CODE REVIEW

File: infrastructure/zk/src/docker.ts [hunk 1]
```diff
@@ -86,6 +90,23 @@ function defaultTagList(image: string, imageTagSha: string, imageTagShaTS: strin
...
+    if (
+        protocolVersionTag &&
+        [
+            'proof-fri-compressor',
+            'proof-fri-gpu-compressor',
+            'prover',
+            'prover-fri',
+            'prover-fri-gateway',
+            'prover-gpu-fri',
+            'prover-v2',
```
Comment (alexandrst88) [inline infrastructure/zk/src/docker.ts:102]: do we need prover-v2 and prover? seems like legacy

**Review by alexandrst88 — CHANGES_REQUESTED** [review]

File: infrastructure/zk/src/docker.ts [hunk 2]
```diff
@@ -66,8 +70,8 @@ async function dockerCommand(
     }
 }
 
-function defaultTagList(image: string, imageTagSha: string, imageTagShaTS: string) {
-    const tagList = [
+function defaultTagList(image: string, imageTagSha: string, imageTagShaTS: string, protocolVersionTag: string) {
+    let tagList = [
```
Comment (alexandrst88) [inline infrastructure/zk/src/docker.ts:74]: any reason you change const -> let?

**Review by alexandrst88 — CHANGES_REQUESTED** [review]

File: infrastructure/zk/src/docker.ts (hunk 1 above)
Comment (yorik) [reply infrastructure/zk/src/docker.ts]: Done

File: infrastructure/zk/src/docker.ts (hunk 2 above)
Comment (yorik) [reply infrastructure/zk/src/docker.ts:74]: To be able to add more tags via push() below.

**Review by Artemka374 — APPROVED** [review]
LGTM

**Review by alexandrst88 — APPROVED** [review]

**Pull request merged**